app/Models/CustomField.php

```diff
@@ -48,7 +48,11 @@ class CustomField extends Model
      *
      * @var array
      */
-    protected $rules = [];
+    protected $rules = [
+        'name' => 'required|unique:custom_fields',
+        'element' => 'required','in:text,listbox,textarea,checkbox,radio',
+        'field_encrypted' => 'nullable|boolean',
+    ];
```

Comment:
I feel like we could add:

```php
'format' => [
                 Rule::in(array_merge(array_keys(self::PREDEFINED_FORMATS), self::PREDEFINED_FORMATS, [$regex_format])),
             ],
      ```
      
      that right into the model-level validation rules, right? Then we could get rid of the `validationRules` method entirely (which I *think* was trying to something else entirely but got transmogrified somehow into subsuming model-level validation...)

Reply:
(Because when we actually *call* `validationRules()` we don't seem to use the parameter at all)

Reply:
https://github.com/snipe/snipe-it/blob/5661e69520851b95dc341f17993fce2c8cab95d0/app/Http/Controllers/Api/CustomFieldsController.php#L103

Reply:
You're right, but I still suspect that we could get rid of the `validationRules` method completely because it's only invoked on Custom Field *creation* (or update). So really just model-level validation will do. The way we validate the *contents* of the custom field (within the asset) is by overriding the `save()` method on the asset - which does *not* call this method.